bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -231,9 +234,15 @@ private URL getUrl(String filename) throws FileNotFoundException {
     private HolidayManager getHolidayManager(Object managerKey) {
         HolidayManager holidayManager = holidayManagers.get(managerKey);
         if (holidayManager == null) {
-            final ManagerParameter parameters = managerKey.getClass() == String.class
-                    ? ManagerParameters.create((String) managerKey)
-                    : ManagerParameters.create((URL) managerKey);
+            final ManagerParameter parameters;
+            if (managerKey.getClass() == String.class) {
```

Comment:
Wouldn't `if (managerKey instanceof String stringKey) {` be better ? This avoids below typecasts

Reply:
Right - fixed. Do you by any chance know how to run the integration tests locally? I would like to refactor a bit more, but first would need to get the tests running.